src/python/pants/base/specs.py

```diff
@@ -192,6 +192,24 @@ def make_glob_patterns(self, address_mapper):
     ]
 
 
+_specificity = {
+  SingleAddress: 0,
+  SiblingAddresses: 1,
+  AscendantAddresses: 2,
+  DescendantAddresses: 3
+}
+
+
+def more_specific(spec1, spec2):
```

Comment:
Type hints appreciated if you know them. I think this would be:

```suggestion
def more_specific(spec1: Spec, spec2: Spec) -> bool:
```